test/modules/returns/controller.js

```diff
@@ -171,12 +184,87 @@ experiment('postUploadPreview', () => {
     expect(msg).to.be.a.string();
 
     expect(params.eventId).to.equal(request.params.eventId);
-    expect(params.companyId).to.equal(request.payload.companyId);
+    expect(params.companyId).to.equal(request.query.companyId);
+  });
+});
+
+experiment('getUploadPreviewReturn', () => {
+  let h;
+
+  const returnServiceResponse = {
+    error: null,
+    data: {
+      metadata: {
+        foo: 'bar'
+      }
+    }
+  };
+
+  const uploadedReturns = [{
+    returnId: 'x',
+    isNil: false,
+    lines: [{
+      quantity: 2
+    }, {
+      quantity: 3
+    }]
+  }, {
+    returnId: 'y',
+    isNil: true
+  }];
+
+  beforeEach(async () => {
+    h = {
+      response: sinon.stub().returns({
+        code: sinon.spy()
+      })
+    };
+    sandbox.stub(logger, 'error');
+    sandbox.stub(uploadValidator, 'validate').resolves(uploadedReturns);
+    sandbox.stub(returnsConnector.returns, 'findOne').resolves(returnServiceResponse);
+  });
+
+  afterEach(async () => {
+    sandbox.restore();
+  });
+
+  test('throws a 404 error if the requested return ID is not in the array', async () => {
+    const request = requestFactory('z');
+    try {
+      await controller.getUploadPreviewReturn(request, h);
```

Comment:
I think we should add a `fail` call here in case in the future the error is not thrown as part of a refactor or something. The test will silently pass in that case.

```
try {
  await controller.getUploadPreviewReturn(request, h);
  fail('this should never happen')
}.....
```